src/ripple/conditions/impl/utils.h

```diff
@@ -175,19 +172,20 @@ parsePreamble(Slice& s, std::error_code& ec)
 
         if (p.length == 0)
         {
-            ec = error::generic;
+            ec = error::malformed_encoding;
             return p;
         }
     }
 
-    ec = {};
+
     return p;
 }
 
 inline
 Buffer
 parseOctetString(Slice& s, std::uint32_t count, std::error_code& ec)
 {
+    // We may want to set a max content size. Do we really want to allow 4 gig messages?
```

Comment:
We really should set _some_ max size now. We can err on the side of way too big, but 4 gig is ridiculous.